internal/db/schema/migrations/postgres/14/01_event.up.sql

```diff
@@ -0,0 +1,217 @@
+│config_id        │     │ 
+│event_type       |     |      
+|sink_type        |     |     ┌─────────────────────┐       
+|format_type      │     │    ╱│ event_sink_type_enm │
+└─────────────────┘     ├─────├─────────────────────┤
+         ┼              │    ╲│                     │
+         │              │     └─────────────────────┘
+         │              │     ┌───────────────────────┐    
+         ○              │    ╱│ event_format_type_enm │    
+        ╱│╲             └─────├───────────────────────┤    
+┌─────────────────┐          ╲│                       │    
+│ event_file_sink │           └───────────────────────┘    
+├─────────────────┤                                  
+│ public_id       │                                  
+│ sink_id         │                                  
+│ path            │                                  
+│ file_name       │                                  
+│ rotate_bytes    │                                  
+│ rotate_duration │                                  
+│ rotate_max_files│                                  
+└─────────────────┘                                                                                      
+*/
+
+create table event_type_enm (
+    name text primary key
+        constraint only_predefined_event_types_allowed
+        check (
+            name in (
+                'every',
+                'error',
+                'audit',
+                'observation',
+                'system'
+            )
+        )
+);
+comment on table event_type_enm is
+'event_type_enm is an enumeration table for the valid event types within '
+'the domain';
+
+create table event_sink_type_enm (
+    name text primary key
+        constraint only_predefined_event_sink_types_allowed
+        check (
+            name in (
+                'stderr',
+                'file'
+            )
+        )
+);
+comment on table event_sink_type_enm is
+'event_sink_type_enm is an enumeration table for the valid sink types within '
+'the domain';
+
+create table event_format_type_enm (
+    name text primary key
+        constraint only_predefined_event_format_types_allowed
+        check (
+            name in (
+                'json',
+                'text',
+                'cloudevent',
+                'hclogtext',
+                'hclogjson'
+            )
+        )
+);
+comment on table event_format_type_enm is
+'event_format_type_enm is an enumeration table for the valid event format types '
+'within the domain';
+
+create table event_config (
+    public_id wt_public_id primary key,
+    scope_id wt_scope_id not null
+        constraint iam_scope_global_fkey
+            references iam_scope_global(scope_id)
+            on delete cascade
+            on update cascade,
+    constraint scope_id_uq -- only allow one config per scope
+        unique (scope_id),   
+    name wt_name,
+    description wt_description,
+    create_time wt_timestamp,
```

Comment:
Just spotted this PR and thought I would take a look, not sure how `create_time` and `update_time` are planned to be set, but if by the DB we will need to add the `before create` and `before update` triggers respectively.